core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -257,6 +272,43 @@ func (c *coordinator) ReportBlocks(
 	return
 }
 
+// getBlockhashesMappingFromRequests returns the blockhashes for enqueued request blocks.
+func (c *coordinator) getBlockhashesMappingFromRequests(
+	ctx context.Context,
+	randomnessRequestedLogs []*vrf_wrapper.VRFBeaconCoordinatorRandomnessRequested,
+	randomnessFulfillmentRequestedLogs []*vrf_wrapper.VRFBeaconCoordinatorRandomnessFulfillmentRequested,
+	currentHeight int64,
+) (blockhashesMapping map[uint64]common.Hash, err error) {
+
+	// Get all request + callback requests into a mapping.
+	rawBlocksRequested := make(map[uint64]struct{})
+	for _, l := range randomnessRequestedLogs {
+		if isBlockEligible(l.NextBeaconOutputHeight, l.ConfDelay, currentHeight) {
+			rawBlocksRequested[l.NextBeaconOutputHeight] = struct{}{}
+		}
+	}
+	for _, l := range randomnessFulfillmentRequestedLogs {
+		if isBlockEligible(l.NextBeaconOutputHeight, l.ConfDelay, currentHeight) {
+			rawBlocksRequested[l.NextBeaconOutputHeight] = struct{}{}
+		}
+	}
+
+	// Fill a unique list of request blocks.
+	requestedBlockNumbers := make([]uint64, len(rawBlocksRequested))
+	i := 0
+	for k := range rawBlocksRequested {
+		requestedBlockNumbers[i] = k
+		i++
```

Comment:
seems like it could be simplified to `requestedBlockNumbers = append(requestedBlockNumbers, k)`